src/studio/src/designer/backend/TypedHttpClients/AltinnStorage/AltinnStorageTextResourceClient.cs

```diff
@@ -0,0 +1,90 @@
+    /// </summary>
+    public class AltinnStorageTextResourceClient : IAltinnStorageTextResourceClient
+    {
+        private readonly HttpClient _httpClient;
+        private readonly PlatformSettings _platformSettings;
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        public AltinnStorageTextResourceClient(
+            HttpClient httpClient,
+            IOptionsMonitor<PlatformSettings> options)
+        {
+            _httpClient = httpClient;
+            _platformSettings = options.CurrentValue;
+        }
+
+        /// <inheritdoc/>
+        public async Task Create(string org, string app, TextResource textResource, EnvironmentModel environmentModel)
+        {
+            Uri uri = CreatePostUri(environmentModel, org, app);
+            AddSubscriptionKeys(uri);
+            string stringContent = JsonSerializer.Serialize(textResource);
+            HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Post, uri)
+            {
+                Content = new StringContent(stringContent, Encoding.UTF8, "application/json"),
+            };
+            await _httpClient.SendAsync(request);
+        }
+
+        /// <inheritdoc/>
+        public async Task<TextResource> Get(string org, string app, string language, EnvironmentModel environmentModel)
+        {
+            Uri uri = CreateGetAndPutUri(environmentModel, org, app, language);
+            AddSubscriptionKeys(uri);
+            HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Get, uri);
+            HttpResponseMessage response = await _httpClient.SendAsync(request);
+            return await response.Content.ReadAsAsync<TextResource>();
+        }
+
+        /// <inheritdoc/>
+        public async Task Update(string org, string app, TextResource textResource, EnvironmentModel environmentModel)
+        {
+            Uri uri = CreateGetAndPutUri(environmentModel, org, app, textResource.Language);
+            AddSubscriptionKeys(uri);
+            string stringContent = JsonSerializer.Serialize(textResource);
+            HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Put, uri)
+            {
+                Content = new StringContent(stringContent, Encoding.UTF8, "application/json"),
+            };
+            await _httpClient.SendAsync(request);
+        }
+
+        private void AddSubscriptionKeys(Uri uri)
+        {
+            if (uri.Host.Contains("tt02", StringComparison.InvariantCultureIgnoreCase))
+            {
+                _httpClient.DefaultRequestHeaders.Add(_platformSettings.SubscriptionKeyHeaderName, _platformSettings.SubscriptionKeyTT02);
+            }
+            else if (uri.Host.Contains("yt01", StringComparison.InvariantCultureIgnoreCase))
+            {
+                _httpClient.DefaultRequestHeaders.Add(_platformSettings.SubscriptionKeyHeaderName, _platformSettings.SubscriptionKeyYT01);
+            }
+        }
```

Comment:
smart extracting this. Could we do it for the appmetadata and policy service as well?

Reply:
Added helper method which is now used by all clients that needs subscription keys.